### PR TITLE
Add coq-category-theory to bench

### PR DESCRIPTION
**Pull request by Janno**
As discussed on Zulip (https://coq.zulipchat.com/#narrow/stream/237656-Coq-devs.20.26.20plugin.20devs/topic/performance.20of.20the.20category_theory.20dev), the category-theory development is unique in the stress it puts on conversion, specifically regarding universe polymorphic terms.

I am not sure if I did this correctly. I suspect somebody with permissions to do so will have to run the `bench` job to find out.

### CODE REVIEW

File: dev/bench/bench.sh [hunk 1]
```diff
@@ -327,6 +327,7 @@ create_opam() {
 
     # Pinning for packages that are not in a repository
     opam pin add -ynq coq-perennial.dev git+https://github.com/mit-pdos/perennial#coq/tested
+    opam pin add -ynq coq-category-theory.dev git+https://github.com/jwiegley/category-theory#master
```
Comment (SkySkimmer) [inline dev/bench/bench.sh:330]: Let's not have more pins. add it to opam-coq-archive instead.

**Review by SkySkimmer — CHANGES_REQUESTED** [review]

File: dev/bench/bench.sh (hunk 1 above)
Comment (Janno) [reply dev/bench/bench.sh:330]: It feels weird to add somebody else's package to the opam archive, even if it is just in extra-dev.

**Comment by Alizter** [comment]
Now that https://github.com/coq/opam-coq-archive/pull/2079 has been merged @Janno, can you update this PR?

**Comment by Janno** [comment]
@Alizter Done! I think just adding the package should suffice now that it's in the opam archive.

**Comment by Alizter** [comment]
@SkySkimmer can you start a bench?

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
bench https://gitlab.com/coq/coq/-/jobs/2017265718

**Comment by Janno** [comment]
The bench finished successfully.
```
┌─────────────────────┬─────────────────────────┬─────────────────────────────────────┬───────────────────────────────────────┬─────────────────────────┬─────────────────┐
│                     │      user time [s]      │             CPU cycles              │           CPU instructions            │  max resident mem [KB]  │   mem faults    │
│                     │                         │                                     │                                       │                         │                 │
│        package_name │     NEW      OLD  PDIFF │           NEW            OLD  PDIFF │            NEW             OLD  PDIFF │     NEW      OLD  PDIFF │ NEW  OLD  PDIFF │
├─────────────────────┼─────────────────────────┼─────────────────────────────────────┼───────────────────────────────────────┼─────────────────────────┼─────────────────┤
│            coq-core │   96.30    96.79  -0.51 │  385748068638   387133577711  -0.36 │   433237288405    433335379814  -0.02 │  272872   273280  -0.15 │   0    0    nan │
│ coq-category-theory │ 1630.63  1626.82   0.23 │ 7454406958656  7437197487864   0.23 │ 15223837683981  15223420788018   0.00 │ 1318008  1318264  -0.02 │   0    0    nan │
│          coq-stdlib │  407.20   400.10   1.77 │ 1649582531769  1622141661361   1.69 │  1409146287401   1409129648763   0.00 │  595856   597848  -0.33 │   0    0    nan │
└─────────────────────┴─────────────────────────┴─────────────────────────────────────┴───────────────────────────────────────┴─────────────────────────┴─────────────────┘
```

**Comment by Alizter** [comment]
Excellent! Merging soon.

**Comment by Alizter** [comment]
@coqbot merge now

**Pull request merged**